test/realm_test.dart

```diff
@@ -1012,5 +1012,29 @@ Future<void> main([List<String>? args]) async {
       //Ensure list size is the same like teams collection size
       expect(list.length, teams.length);
     });
+
+    test('Realm deleteRealm files after closing realm', () {
+      var config = Configuration([Dog.schema, Person.schema]);
+      var realm = Realm(config);
+      
+      realm.close();
+      Realm.deleteRealm(config);
+
+      expect(File(config.path).existsSync(), false);
+      // TODO: For Discussion: Lock file is not deleted by core API.
+      //expect(File("${config.path}.lock").existsSync(), false);
+      expect(Directory("${config.path}.management").existsSync(), false);
+    });
+
+    test('Realm deleteRealm files before closing realm expects exception', () {
+      var config = Configuration([Dog.schema, Person.schema]);
+      var realm = Realm(config);
+
+      expect(() => Realm.deleteRealm(config), throws<RealmException>());
+      
+      expect(File(config.path).existsSync(), true);
+      expect(File("${config.path}.lock").existsSync(), true);
```

Comment:
imo, we should not check for the lock file in the test.

Reply:
Fixed in https://github.com/realm/realm-dart/pull/169/commits/0697dce4c21edbd697003715971e4e21b9d68d23